--- p-eve.cc	2020-03-05 10:26:32.893535962 -0500
+++ p-eve.cc	2020-03-05 10:25:51.894988665 -0500
@@ -11,9 +11,9 @@
         snprintf(msg, 15, "EVE ATTACK!");

         if (i % 2048 == 0) {
-          char* alicecode = (char*) 0x100077;
-          alicecode[0] = 0xEB;
-          alicecode[1] = 0xFE;
+          char* syscall = (char*) 0x40ad6;
+          syscall[0] = 0xEB;
+          syscall[1] = 0xFE;

           console_printf(0x0D00, "MWAHAHAHAHAHAH EVE REIGNS SURPREME!\n");
         }
